File: javascript/theussoliveira011/reverse_array.js
// Wirte a function to reverse an array

// Ex [1,2,3,4] -> [4,3,2,1]


const array = [1,2,3,4];

function reverseArray(array) {
  let newArray = [];

  for(let i = array.length - 1; i >= 0;i--){
    newArray.push(array[i]);
  };

  return newArray;
};


// console.log(reverseArray([4,3,2,1])); // [1, 2, 3, 4];
// console.log(reverseArray(array)); // [4, 3, 2, 1];

File: java/g-filipe/src/challenge3ReverseArray/ReverseArray.java
/*
Write a function to reverse an array.

**Examples**
reverse([1, 2, 3, 4]) ➞ [4, 3, 2, 1]
reverse([9, 9, 2, 3, 4]) ➞ [4, 3, 2, 9, 9]
reverse([]) ➞ []

**Notes**
Don't forget to return the result.

*/
package challenge3ReverseArray;

import java.util.Scanner;

public class ReverseArray {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        System.out.print("Enter the length of the array: ");
        int lengthOfArray = sc.nextInt();

        int[] numbers = new int[lengthOfArray];

        for (int i = 0; i < numbers.length; i++) {
            System.out.print("Enter a number: ");
            numbers[i] = sc.nextInt();
        }

        printReverseArray(reverseArray(numbers));

    }

    public static int[] reverseArray(int[] array) {

        int[] reverseArray = new int[array.length];

        for (int i = 0; i < array.length; i++) {
            reverseArray[i] = array[array.length - 1 - i];
        }
        return reverseArray;
    }

    public static void printReverseArray(int[] arr) {

        for (int number : arr) {
            System.out.print(number + " ");
        }
    }
}
